docs: update Quickstart.mdx, cleanup (#2059)
enarx/enarx: docs/Running/Publish.md

# Publish & Deploy Apps

> Enarx is more than just a tool for running WebAssembly. It also allows you to publish a WebAssembly package to a remote package host, as well as securely run a published package in a local Enarx keep. The following sections will demonstrate this process, along with other useful commands for interacting with a package host.

Reviewer: Why not just call it a Drawbridge registry with a reference to the project?
"package host" is very ambiguous, is that an SSH host? Is that an NFS share? Is that some custom service? People reading this should not be asking themselves this kind of questions

Reviewer 2: I'm using "package host" specifically to avoid introducing the jargon "drawbridge", which is even less descriptive unless first explained. If we actually want to use "drawbridge" as a term then we can go ahead and define it here, but that was yet to be determined when this document was written.

## Registering a user and authenticating with the package host

Before we begin, you will need a GitHub account in order to authenticate to the package host.

First, register a new user with the package host using the `enarx user register` command, as shown here:

```
enarx user register your_username
```

> Upon entering this command, Enarx will print two things to your terminal: a web page URL, and a unique one-time code. Open the URL, then copy and paste the one-time code into the form field on the web page. Once you have entered the code, it will ask to connect to your GitHub account. Follow the instructions shown on the page, and when you have finished connecting your GitHub account you may close the page and return to your terminal. Note that it may take a few moments before your terminal finishes authenticating with GitHub. Once it does, it will notify you that your credentials have been saved locally. The command will then exit, indicating that your user account has been successfully registered.

Reviewer: We should not here that the link should can be opened on a different device than the one on which the login command is run, e.g. when logging in on remote, headless hosts

Reviewer 2: Done.

Note that in the future when you need to log in again (for example, if you are connecting from a different computer), you can do so with the `enarx user login` command, as shown here:

```
enarx user login
```

## Registering a new repository

> Before you can publish a package, you will first have to register the repository with the package host. You can do so with the `enarx repo register` command, as shown here:

Reviewer: I think this should really just be create and we probably want to rename the CLI command accordingly.
Other projects users are already familiar (GitHub, Docker Hub) use the same terminology

Reviewer 2: Bikeshedding names can be tackled separately, please file an issue.

```
enarx repo register your_username/your_reponame
```

In the above example, `your_username` is the username that you registered previously, and `your_reponame` is the name that you want this repository to have.

## Publishing a WebAssembly package

Before you can publish, you will first need to [compile your application to WebAssembly](../WebAssembly/Introduction). At the end of this process you will have a file with the `.wasm` file extension. Rename this file to `main.wasm` and place it in the same directory as a properly configured [`Enarx.toml`](Enarx_toml).

Once you have a directory containing a `main.wasm` and an `Enarx.toml`, we can *publish* this directory to the package host with the `enarx package publish` command, as shown here:

```
enarx package publish your_username/your_reponame:0.1.0 your_directory
```

In the above example, `0.1.0` is a *tag*, which identifies a unique version of the package being uploaded to this repository.

## Running a published package

Once a package has been published, it can be run directly with the `enarx deploy` command, as shown here:

```

> enarx deploy some_username/some_reponame:0.1.0

Reviewer: why suddenly use different names here?

Reviewer 2: The use of `your_username` vs. `some_username` is supposed to emphasize that you can deploy anyone's (public) package, not just one that you own.

```

> Unlike `enarx repo register` and `enarx package publish`, this command does not require authentication and can be used with any public package.

Reviewer: ```suggestion
Unlike `enarx repo register` and `enarx package publish`, this command does not require authentication and can deploy any public package.
```

Reviewer 2: Done, but note that last I asked Nathaniel he still wanted us to get rid of `enarx deploy` as a concept and just have a unified `enarx run` command, so I don't know whether being consistent in our jargon here will matter in the long run.
